src/kernel_module.cpp

```diff
@@ -54,7 +54,9 @@ Value KernelModule::binding(Env *env) {
 }
 
 Value KernelModule::caller(Env *env) {
-    auto ary = env->backtrace();
+    auto backtrace = env->backtrace();
+    auto ary = backtrace->to_ruby_array();
+    backtrace->mark();
```

Comment:
I'm not sure if this is a great way to do this. Basically I want to tell the GC that this object can be sweeped at any time from there on. Is this the way to do this? (Maybe we should rethink allocating a GC-managed Backtrace object in this case?)

Reply:
Actually, that has the opposite effect, but only for the next mark/sweep cycle. "Mark" in the GC means "this object is in use" (during this cycle). Once the GC mark/sweep cycle is over, all objects get unmarked. And a subsequent cycle should clean it up.

Probably best to just remove the `mark()` line here and let the GC do its job. :smile:

Reply:
Oops, I'm going to remove it 😅